plugin/evm/gossip.go

```diff
@@ -19,11 +22,65 @@ import (
 )
 
 var (
+	_ p2p.Handler = (*txGossipHandler)(nil)
+
 	_ gossip.Gossipable        = (*GossipEthTx)(nil)
 	_ gossip.Gossipable        = (*GossipAtomicTx)(nil)
 	_ gossip.Set[*GossipEthTx] = (*GossipEthTxPool)(nil)
 )
 
+func newTxGossipHandler[T any, U gossip.GossipableAny[T]](
+	log logging.Logger,
+	mempool gossip.Set[U],
+	metrics gossip.Metrics,
+	maxMessageSize int,
+	throttlingPeriod time.Duration,
+	throttlingLimit int,
+	validators *p2p.Validators,
+) txGossipHandler {
+	// push gossip messages can be handled from any peer
+	handler := gossip.NewHandler[T, U](
+		log,
+		// Don't forward gossip to avoid double-forwarding while legacy
+		// gossip is still running.
+		gossip.NoOpAccumulator[U]{},
```

Comment:
So I think for now we can't actually use the push gossiper to forward transactions. The mempool gets notified of a new tx by looking at a channel which is updated on add, but since this handler adds the new txs to the mempool this will result in a double gossip if we don't skip the push gossip step after adding to the mempool.